rmgpy/rmg/mainTest.py

```diff
@@ -183,6 +184,78 @@ def test_make_cantera_input_file(self):
                     self.fail('The output Cantera file is not loadable in Cantera.')
 
 
+@attr('functional')
+class TestRestartWithFilters(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """A function that is run ONCE before all unit tests in this class."""
+        cls.testDir = os.path.join(originalPath, 'rmg', 'test_data', 'restartTest')
+        cls.outputDir = os.path.join(cls.testDir, 'output_w_filters')
+        cls.databaseDirectory = settings['database.directory']
+
+        os.mkdir(cls.outputDir)
+        initialize_log(logging.INFO, os.path.join(cls.outputDir, 'RMG.log'))
+
+        cls.rmg = RMG(input_file=os.path.join(cls.testDir, 'restart_w_filters.py'),
+                      output_directory=os.path.join(cls.outputDir))
+
+        cls.rmg.execute()
+
+    def test_restart_with_filters(self):
+        """
+        Test that the RMG restart job with filters included completed without problems
+        """
+        with open(os.path.join(self.outputDir, 'RMG.log'), 'r') as f:
+            self.assertIn('MODEL GENERATION COMPLETED', f.read())
+
+    @classmethod
+    def tearDownClass(cls):
+        """A function that is run ONCE after all unit tests in this class."""
+        # Reset module level database
+        import rmgpy.data.rmg
+        rmgpy.data.rmg.database = None
+
+        # Remove output directory
+        shutil.rmtree(cls.outputDir)
+
+
+@attr('functional')
+class TestRestartNoFilters(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """A function that is run ONCE before all unit tests in this class."""
+        cls.testDir = os.path.join(originalPath, 'rmg', 'test_data', 'restartTest')
+        cls.outputDir = os.path.join(cls.testDir, 'output_no_filters')
+        cls.databaseDirectory = settings['database.directory']
+
+        os.mkdir(cls.outputDir)
+        initialize_log(logging.INFO, os.path.join(cls.outputDir, 'RMG.log'))
+
+        cls.rmg = RMG(input_file=os.path.join(cls.testDir, 'restart_no_filters.py'),
+                      output_directory=os.path.join(cls.outputDir))
+
+        cls.rmg.execute()
```

Comment:
I feel like executing the job should be part of the test method rather than in setup.

Reply:
I agree. Sorry, I copied from the tests above, but I agree this makes more sense.